bigtable/tests/mutations_integration_test.cc

```diff
@@ -0,0 +1,511 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/testing/table_integration_test.h"
+
+namespace {
+
+namespace admin_proto = ::google::bigtable::admin::v2;
+
+class MutationIntegrationTest : public bigtable::testing::TableIntegrationTest {
+ protected:
+  std::string const column_family1 = "fam1";
+  std::string const column_family2 = "fam2";
+  std::string const column_family3 = "fam3";
+
+  bigtable::TableConfig table_config = bigtable::TableConfig(
+      {{column_family1, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family2, bigtable::GcRule::MaxNumVersions(10)},
+       {column_family3, bigtable::GcRule::MaxNumVersions(10)}},
+      {});
+
+  /**
+   * This function creates Cell by ignoring the timestamp.
+   * In this case Cloud Bigtable will insert the default server
+   * side timestamp for the cells.
+   */
+  void CreateCellsIgnoringTimestamp(bigtable::Table& table,
+                                    std::vector<bigtable::Cell> const& cells) {
+    std::map<std::string, bigtable::SingleRowMutation> mutations;
+    for (auto const& cell : cells) {
+      std::string key = cell.row_key();
+      auto inserted = mutations.emplace(key, bigtable::SingleRowMutation(key));
+      inserted.first->second.emplace_back(bigtable::SetCell(
+          cell.family_name(), cell.column_qualifier(), cell.value()));
+    }
+
+    bigtable::BulkMutation bulk;
+    for (auto& kv : mutations) {
+      bulk.emplace_back(std::move(kv.second));
+    }
+    table.BulkApply(std::move(bulk));
+  }
+
+  /**
+   * Delete the records from table for the specified family and column
+   * identifier.
+   */
+  void DeleteFromColumn(bigtable::Table& table, std::string row_key,
+                        std::string column_family, std::string column_id) {
+    bigtable::SingleRowMutation mutation(row_key);
+    mutation.emplace_back(bigtable::DeleteFromColumn(column_family, column_id));
+
+    bigtable::BulkMutation bulk;
```

Comment:
These (here and in some of the following functions), could be replaced with `table.Apply(mutation)`, since you are changing a single row.  You could even do:

```C++
table.Apply(bigtable::SingleRowMutation(row_key, bigtable::DeleteFromColumn(column_family, colum_id)));
```

I do not believe it is that important, but I thought I would mention it for future reference.